pennylane/capture/transforms/capture_decompose.py

```diff
@@ -0,0 +1,126 @@
+        else:
+            self.gate_set = gate_set
+
+        super().__init__()
+
+    def stopping_condition(self, op: qml.operation.Operator) -> bool:
+        """Function to determine whether or not an operator needs to be decomposed or not.
+
+        Args:
+            op (qml.operation.Operator): Operator to check.
+
+        Returns:
+            bool: Whether or not ``op`` is valid or needs to be decomposed. ``True`` means
+            that the operator does not need to be decomposed.
+        """
+        if not op.has_decomposition:
+            if not self.gate_set(op):
+                warnings.warn(
+                    f"Operator {op.name} does not define a decomposition and was not "
+                    f"found in the target gate set. To remove this warning, add the operator name "
+                    f"({op.name}) or type ({type(op)}) to the gate set.",
+                    UserWarning,
+                )
+            return True
+        return self.gate_set(op)
+
+    def decompose_operation(self, op: qml.operation.Operator):
+        """Decompose a PennyLane operation instance if it does not satisfy the
+        provided gate set.
+
+        Args:
+            op (Operator): a pennylane operator instance
+
+        Returns:
+            Any
+
+        This method is only called when the operator's output is a dropped variable,
+        so the output will not affect later equations in the circuit.
+
+        See also: :meth:`~.interpret_operation_eqn`, :meth:`~.interpret_operation`.
+        """
+        if self.gate_set(op):
+            return self.interpret_operation(op)
+
+        qml.capture.disable()
+        try:
+            decomposition = list(
+                _operator_decomposition_gen(
+                    op, self.stopping_condition, max_expansion=self.max_expansion
+                )
+            )
+         finally:
+            qml.capture.enable()
+
+        return [self.interpret_operation(decomp_op) for decomp_op in decomposition]
+
+    def interpret_operation_eqn(self, eqn):
+        """Interpret an equation corresponding to an operator.
+
+        Args:
+            eqn (jax.core.JaxprEqn): a jax equation for an operator.
+
+        See also: :meth:`~.interpret_operation`.
+
+        """
+        invals = (self.read(invar) for invar in eqn.invars)
+        with qml.QueuingManager.stop_recording():
+            op = eqn.primitive.impl(*invals, **eqn.params)
+        if eqn.outvars[0].__class__.__name__ == "DropVar":
+            return self.decompose_operation(op)
+        return op
+
+
+@DecomposeInterpreter.register_primitive(ctrl_transform_prim)
+def handle_ctrl_transform(*_, **__):  # pylint: disable=missing-function-docstring
+    raise NotImplementedError
+
+
+@DecomposeInterpreter.register_primitive(adjoint_transform_prim)
+def handle_adjoint_transform(*_, **__):  # pylint: disable=missing-function-docstring
+    raise NotImplementedError
```

Comment:
Do we want to just decompose the contents of the adjoint?  Does the default `PlxprInterpreter` logic do hat?

Reply:
I'm trying to think if there's any instance in which `adjoint(op).decomposition()` won't just be the same as `reversed([adjoint(o) for o in op.decomposition()])`. If that's the case, then probably best to leave `adjoint` unimplemented for now.

Reply:
> Does the default `PlxprInterpreter` logic do hat?

Yes it does.

Reply:
I updated adjoint to just work.